### DISCUSSION TITLE
Any idea of how to detect the last element in an "each" with an internal "if"

### DISCUSSION THREAD

**Discussion by jpeter06** [mockoon/mockoon#1039] [Help] [answered]
Hello All,
I am using **#each**, to iterate througt a list.
Aditionally I am filtering the list with an inner **#if** 

Problem: only when de last element is included in the filter the "," is not inserted.
@last cannot be used.
Example of the code

```
[ 
  {{setVar 'type' (queryParam 'type' '') }}
  {{#each (dataRaw 'zones' )}} 
    {{#if (includes this.type @type)}}
      {"label": "{{this.label}}", "value": "{{this.value}}"}
      {{#unless @last}},{{/unless}}
    {{/if}}
  {{/each}}
]
```

Thanks in advance :-)

**Comment by 255kb** [comment, marked as answer]
Hi, 

The best solution would be to have a `filter` helper to create an array and then do an `each`. But we don't have such helper. You can open an issue if you want. 

I still found an alternative by creating another variable counting the number of item found: 

```
[ 
  {{setVar 'type' (queryParam 'type' '') }}

  {{setVar 'nb' 0}}

  {{#each (dataRaw 'zones' )}} 
    {{#if (includes this.type @type)}}

    {{setVar 'nb' (add (getVar 'nb') 1)}}

    {{#if (gt @nb 1)}}
      {{#unless @../last}},{{/unless}}
    {{/if}}

    {"label": "{{this.label}}", "value": "{{this.value}}"}

    {{/if}}
  {{/each}}
]
```

**Reply by jpeter06** [reply]
Thanks Guillaume,
I will ask for this "filter" helper. And maybe more things  :-D
At the end I finished using a boolean variable:

**{{setVar 'addComma' false }}**
{{setVar 'extKey' (queryParam 'extKey' '') }}
[ 
  {{#each (dataRaw 'op_zonas' )}} 
    {{#if (includes this.extKey @extKey)}}
      **{{#if @addComma}},{{/if}}**
      {"label": "{{this.label}}", "value": "{{this.value}}"}
      **{{setVar 'addComma' true }}**
    {{/if}}
  {{/each}}
]
